### ISSUE TITLE
kotlin false positive, JXI_UNDEFINED_PARAMETER_SOURCE_IN_ENDPOINT

### ISSUE THREAD

**Issue by askbeka** [spotbugs/spotbugs#1165]
maven spotbugs plugin is being trigereed 
```
[INFO] --- spotbugs-maven-plugin:4.0.0:check (default) @ my-api ---
[INFO] BugInstance size is 1
[INFO] Error size is 0
[INFO] Total bugs: 1
[ERROR] JAX-RS Method com.my.api.MyResource.postStuff(RequestBody) defines a parameter that has no @*Param or @Context annotation, or @Consumes method annotation JXI_UNDEFINED_PARAMETER_SOURCE_IN_ENDPOINT
```
but Resource does have `@Consumes` annotation like:
```kt
import javax.validation.Valid
import javax.ws.rs.Consumes
import javax.ws.rs.POST
import javax.ws.rs.Path
import javax.ws.rs.Produces
import javax.ws.rs.core.MediaType
import javax.ws.rs.core.Response

@Path("/v1/post-stuff")
class MyResource {
    @POST
    @Consumes(MediaType.APPLICATION_JSON)
    @Produces(MediaType.APPLICATION_JSON)
    fun postStuff(@Valid requestBody: RequestBody) =
            Response
                    .status(Response.Status.OK)
                    .entity(ResponseBody("posted"))
                    .build()
}

```

**Comment by ThrawnCA** [comment]
I'm guessing this is a matter of how Kotlin generates bytecode.

**Comment by askbeka** [comment]
Sure. Kotlin does generate code for jvm. But does it mean that kotlin
generates buggy code?

**Comment by ThrawnCA** [comment]
Not necessarily buggy, but it might move things around such that SpotBugs can no longer recognise the placement of the annotation.

**Comment by askbeka** [comment]
@ThrawnCA can you think of the way t resolve the issue?
I am not very familiar with technologies and the way plugin is built. Can do some investigation with proper direction
